core/lib/dal/src/consensus_dal.rs

```diff
@@ -412,11 +470,12 @@ impl ConsensusDal<'_, '_> {
     ) -> anyhow::Result<()> {
         let res = sqlx::query!(
             r#"
-            INSERT INTO
-                l1_batches_consensus (l1_batch_number, certificate, created_at, updated_at)
-            VALUES
-                ($1, $2, NOW(), NOW())
-            ON CONFLICT (l1_batch_number) DO NOTHING
+            UPDATE l1_batches_consensus
+            SET
+                certificate = $2,
+                updated_at = NOW()
+            WHERE
+                l1_batch_number = $1
```

Comment:
On second thought this shouldn't happen: before the certificate for batch N can be verified the node needs the committee from the execution of batch N-1, so I assume there is plenty of time between when the committee is extracted from N-1 and inserted for N, especially because we need the commitment for batch N too. 

Still it feels like having two tables would avoid any confusion.